src/main/webapp/src/app/user-profile/user-profile.component.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {HttpClient, HttpResponse} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {User} from '../model/user.model';
+
+@Component({
+  selector: 'app-user-profile',
+  templateUrl: './user-profile.component.html',
+  styleUrls: [
+    './user-profile.component.css',
+    '../common/growpod-page-styles.css',
+  ],
+})
+
+/**
+ * This is a component designed to display a user profile, either the
+ * logged in user or another user.
+ *
+ * This page takes an argument 'id', the id of the user it requests.
+ */
+export class UserProfileComponent implements OnInit {
+  displayInfo: User | null;
+  errorMessage = '';
+
+  /**
+   * Initializes the component based on provided arguments
+   *
+   * @param route Contains arguments.
+   */
+  constructor(private route: ActivatedRoute, private httpClient: HttpClient) {
+    const idArg = route.snapshot.paramMap.get('id');
+    const id = idArg === null ? 'current' : idArg;
```

Comment:
```suggestion
    const id = idArg ?? 'current';
```

Reply:
Any interest in guarding against the empty string? In that case `||` also works.

Reply:
Will change to `??`. No interest in guarding against `""`, will make clear in annotations.